### PR TITLE
parser,ddl: support altering the comment of a sequence

### LINKED ISSUE #51433: Can't alter the COMMENT on a SEQUENCE
## Feature Request

Would be great to be able to change the comment on a sequence.

```
sql> CREATE SEQUENCE s1 COMMENT "seq 1";
Query OK, 0 rows affected (0.0261 sec)

sql> CREATE SEQUENCE s2;
Query OK, 0 rows affected (0.0125 sec)

sql> ALTER SEQUENCE s2 COMMENT "seq 2";
ERROR: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 25 near "COMMENT "seq 2"" 

sql> TABLE INFORMATION_SCHEMA.SEQUENCES;
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------+
| TABLE_CATALOG | SEQUENCE_SCHEMA | SEQUENCE_NAME | CACHE | CACHE_VALUE | CYCLE | INCREMENT | MAX_VALUE           | MIN_VALUE | START | COMMENT |
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------+
| def           | test            | s1            |     1 |        1000 |     0 |         1 | 9223372036854775806 |         1 |     1 | seq 1   |
| def           | test            | s2            |     1 |        1000 |     0 |         1 | 9223372036854775806 |         1 |     1 |         |
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------+
2 rows in set (0.0005 sec)

sql> SHOW CREATE SEQUENCE s1\G
*************************** 1. row ***************************
       Sequence: s1
Create Sequence: CREATE SEQUENCE `s1` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB COMMENT='seq 1'
1 row in set (0.0011 sec)

```

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #51433

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
It is now possible to change the comment of a sequence with a `ALTER SEQUENCE` statement.
```

### CODE REVIEW

**Comment by dveeden** [comment]
```
sql> CREATE SEQUENCE s1 COMMENT 'original comment';
Query OK, 0 rows affected (0.0131 sec)

sql> ALTER SEQUENCE s1 COMMENT 'new updated comment';
Query OK, 0 rows affected (0.0113 sec)

sql> SHOW CREATE SEQUENCE s1\G
*************************** 1. row ***************************
       Sequence: s1
Create Sequence: CREATE SEQUENCE `s1` start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB COMMENT='new updated comment'
1 row in set (0.0004 sec)

sql> SELECT * FROM information_schema.sequences;
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------------------+
| TABLE_CATALOG | SEQUENCE_SCHEMA | SEQUENCE_NAME | CACHE | CACHE_VALUE | CYCLE | INCREMENT | MAX_VALUE           | MIN_VALUE | START | COMMENT             |
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------------------+
| def           | test            | s1            |     1 |        1000 |     0 |         1 | 9223372036854775806 |         1 |     1 | new updated comment |
+---------------+-----------------+---------------+-------+-------------+-------+-----------+---------------------+-----------+-------+---------------------+
1 row in set (0.0005 sec)

sql> 
```

File: pkg/parser/parser.y [hunk 1]
```diff
@@ -15109,14 +15109,24 @@ DropSequenceStmt:
  *	[ RESTART [WITH | = ] restart ]
  ********************************************************************************************/
 AlterSequenceStmt:
-	"ALTER" "SEQUENCE" IfExists TableName AlterSequenceOptionList
+	"ALTER" "SEQUENCE" IfExists TableName AlterSequenceOptionList PartDefOptionList
```
Comment (AilinKid) [inline pkg/parser/parser.y:15112]: why not use `CreateTableOptionListOpt`, which is used in createSequence

File: pkg/ddl/sequence.go [hunk 2]
```diff
@@ -220,6 +220,16 @@ func alterSequenceOptions(sequenceOptions []*ast.SequenceOption, ident ast.Ident
 			restartValue = op.IntValue
 		}
 	}
+	for _, op := range tableOptions {
+		switch op.Tp {
+		case ast.TableOptionComment:
+			oldSequence.Comment = op.StrValue
+		case ast.TableOptionEngine:
```
Comment (AilinKid) [inline pkg/ddl/sequence.go:227]: previously, TiDB implement self sequence according what Mariadb does
```
create sequence seq;
alter sequence seq comment="haha";
ERROR 1064 (42000) at line 3: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'comment="haha"' at line 1
```
And seems Mariadb will error about this, is it reasonable?

**Review by AilinKid — COMMENTED** [review]
Rest is fine to me

File: pkg/ddl/sequence.go (hunk 2 above)
Comment (dveeden) [reply pkg/ddl/sequence.go:227]: That's indeed a valid concern.

However I've noticed some more differences between our implementation and the one from MariaDB:
- `SHOW CREATE TABLE <sequence>` has completely different output.

Comment (dveeden) [reply pkg/ddl/sequence.go:227]: PostgreSQL allows updating the comment, but the syntax is completely different.
```
postgres=# CREATE SEQUENCE s1;
CREATE SEQUENCE
postgres=# COMMENT ON SEQUENCE s1 IS 'Example comment';
COMMENT
postgres=# \d+
                                        List of relations
 Schema | Name |   Type   |  Owner   | Persistence | Access method |    Size    |   Description   
--------+------+----------+----------+-------------+---------------+------------+-----------------
 public | s1   | sequence | postgres | permanent   |               | 8192 bytes | Example comment
(1 row)

postgres=# COMMENT ON SEQUENCE s1 IS 'Updated sequence comment';
COMMENT
postgres=# \d+
                                             List of relations
 Schema | Name |   Type   |  Owner   | Persistence | Access method |    Size    |       Description        
--------+------+----------+----------+-------------+---------------+------------+--------------------------
 public | s1   | sequence | postgres | permanent   |               | 8192 bytes | Updated sequence comment
(1 row)

postgres=# SELECT VERSION();
                                                      version                                                      
-------------------------------------------------------------------------------------------------------------------
 PostgreSQL 15.6 (Debian 15.6-0+deb12u1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
(1 row)
```

**Comment by CbcWestwolf** [comment]
The code looks fine for me. 
Since this PR changes the behavior of SEQUENCE, maybe a related PR on https://github.com/pingcap/docs or more detail decription in https://github.com/pingcap/tidb/issues/51433 is needed to clarify this change.

**Comment by dveeden** [comment]
Closing this as we follow the MariaDB bahavior for now

**Pull request closed without merging**